source/highlight.c

```diff
@@ -0,0 +1,146 @@
+#include <stdlib.h>
+#include <string.h>
+#include <helper.h>
+
+#include <highlight.h>
+#include <buffer.h>
+#include <screen.h>
+
+
+static char* get_file_extension( const char *filename ) {
+    return strrchr(filename, '.') + 1;
+}
+
+static int str_empty( char *str ) {
+    return str[0] == '\0';
```

Comment:
What if `str = nil` or `str = ""`? :boom:

Reply:
The utility function `str_empty` isn't needed anymore since the most recent commits, and was therefore removed